src/backend/distributed/utils/statistics_collection.c

```diff
@@ -240,6 +411,71 @@ SendHttpPostJsonRequest(const char *url, const char *jsonObj, long timeoutSecond
 }
 
 
+/*
+ * SendHttpGetRequest sends an HTTP/HTTPS GET request to the given URL. GET parameters
+ * should be added to the url. responseCallback is called with the content of response.
+ */
+static bool
+SendHttpGetRequest(const char *url, long timeoutSeconds,
+				   curl_write_callback responseCallback)
+{
+	bool success = false;
+	CURL *curl = NULL;
+
+	curl_global_init(CURL_GLOBAL_DEFAULT);
+	curl = curl_easy_init();
+	if (curl)
+	{
+		curl_easy_setopt(curl, CURLOPT_URL, url);
+		curl_easy_setopt(curl, CURLOPT_TIMEOUT, timeoutSeconds);
+		curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, responseCallback);
+
+		success = PerformHttpRequest(curl);
+
+		curl_easy_cleanup(curl);
+	}
+
+	curl_global_cleanup();
+
+	return success;
+}
+
+
+/*
+ * PerformHttpRequest sends the HTTP request with the parameters set in the given
+ * curl object, and returns if it was successful or not. If the request was not
+ * successful, it may log some warnings.
+ */
+static bool
+PerformHttpRequest(CURL *curl)
```

Comment:
Moved the error checking code to here to avoid much repetition between GET/POST functions.

Reply:
Done.